universal-application-tool-0.0.1/app/repository/AmazonS3Client.java

```diff
@@ -103,9 +109,27 @@ private void getTestObject() {
 
   private void connect() {
     String regionName = config.getString(AWS_S3_REGION);
+    String endpoint;
     region = Region.of(regionName);
     bucket = config.getString(AWS_S3_BUCKET);
+    String dev_env = config.getString(DEV_ENV);
+    URI endpointOverride;
+
+    endpoint = config.getString(AWS_LOCAL_ENDPOINT);
+    try {
+        if (dev_env.equals("1")) {
+            endpointOverride = new URI(endpoint);
+            s3 = S3Client.builder().endpointOverride(endpointOverride).region(region).build();
 
-    s3 = S3Client.builder().region(region).build();
+        } else {
+            s3 = S3Client.builder().region(region).build();
+        }
+
+    } catch(URISyntaxException e) {
```

Comment:
Suggestion (blocking): please have a narrower try-catch scope. It is okay to throw a RuntimeException here as this is only used in dev mode.

```
S3ClientBuilder s3ClientBuilder = S3Client.builder().region(region);
if (environment.isDev()) {
  try {
    URI localEndpoint = new URI(config.getString(AWS_LOCAL_ENDPOINT));
    s3ClientBuilder = s3ClientBuilder.endpointOverride(localEndpoint);
  } catch (URISyntaxException e) {
    throw new RuntimeException(e);
  }
}
s3 = s3ClientBuilder.build();
```